ntbs-service/Models/Notification.cs

```diff
@@ -1,22 +1,33 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
 using System.Linq;
+using ExpressiveAnnotations.Attributes;
 using ntbs_service.Models.Enums;
+using ntbs_service.Models.Validations;
 
 namespace ntbs_service.Models
 {
-    public class Notification
+    public class Notification : ModelBase
     {
         public Notification() {
             NotificationStatus = NotificationStatus.Draft;
+            PatientDetails = new PatientDetails();
+            Episode = new Episode();
+            SocialRiskFactors = new SocialRiskFactors();
+            ClinicalDetails = new ClinicalDetails();
+            PatientTBHistory = new PatientTBHistory();
+            ContactTracing = new ContactTracing();
         }
-        
+
         public int NotificationId { get; set; }
         public DateTime? SubmissionDate { get; set; }
         public NotificationStatus NotificationStatus { get; set; }
         
         public virtual PatientDetails PatientDetails { get; set; }
         public virtual ClinicalDetails ClinicalDetails { get; set; }
+
+        [AssertThat("(NotificationSites.Count > 0 && ShouldValidateFull) || !ShouldValidateFull", ErrorMessage = ValidationMessages.DiseaseSiteIsRequired)]
```

Comment:
Not sure `ShouldValidateFull` wants to be in this logic twice?

Reply:
Second one is !ShouldValidateFull. List<NotificationSites> is an empty list and therefore RequiredIf is not working on it

Reply:
Hmm, still not convicnced, we've got
`(A && B) || !B)`
-> `!(!A || !B) || !B`
-> `!((!A || !B) && B)`
-> `!((!A && B) || (!B && B))`
-> `!((!A && B) || 0)`
-> `!(!A && B)`
-> `A || !B`

So I think it should just read `(NotificationSites.Count > 0 || !ShouldValidateFull)`, which is the boolean talk for "if validate in full, then make sure count is more than 0"

Reply:
Ah I see what you mean